### ISSUE TITLE
curl, sftp, ssh and other commands not working

### ISSUE THREAD

**Issue by louisdh** [louisdh/terminal#99]
@holzschu Is there a bug in ios_system causing these command to not work in OpenTerm?

To be clear: the commands are shown in auto complete, but trying to use them results in:
`command not found`

**Comment by holzschu** [comment]
I can't reproduce the bug (meaning they work on my side). I have a fresh clone of OpenTerm (cloned this morning). 

`command not found` when the command show up in auto-complete is most likely the dynamic library is not embedded in the application (that would be `libcurl.dylib` and `libssh_cmd.dylib`). 

But these libraries *are* in the Xcode project, so it's not that. It could be that these libraries are missing a supplementary dynamic library they need to load. Like `libssh2.framework` and `openssl.framework`. These need to be both linked with iOS_system and embedded in the application.

Try addind these two to the embedded binaries. 
Defining SIDELOADING can give more debug information on the dynamic library loading phase.

**Comment by louisdh** [comment]
Defining SIDELOADING provides the following error:
```
Failed loading curl from libcurl.dylib, cause = dlopen(libcurl.dylib, 9): Library not loaded: @rpath/openssl.framework/openssl
  Referenced from: /var/containers/Bundle/Application/CD9FA560-679F-4483-8595-803811D47EA0/OpenTerm.app/Frameworks/libcurl.dylib
  Reason: image not found
Failed loading curl from libcurl.dylib, cause = invalid handle passed to dlsym()
```

**Comment by holzschu** [comment]
Which confirms my main hypothesis: `libcurl.dylib` is present in the list of "Embedded frameworks", so the system tries to load it. But when it loads it, it needs to load all the dynamic libraries it depends on, namely `openssl.framework` and `libss2.framework`.  These two are not in "Embedded frameworks", so they can't be loaded, and loading `libcurl.dylib` fails.

Add `openssl.framework` and `libss2.framework` to the list of "Embed frameworks" in "Build phase".

**Comment by louisdh** [comment]
I can reproduce this issue with a fresh clone:
* `git clone git@github.com:louisdh/terminal.git openterm`
* `./bootstrap.sh`
* run on iPad

**Comment by louisdh** [comment]
@holzschu Since this does work on your end, do you happen to have any uncommitted changes?

**Comment by holzschu** [comment]
I have no uncommitted changes, but the dynamic libraries, once installed, stay installed in the Application folder on the iPad. 

Solution: open OpenTerm.xcworkspace, go to OpenTerm, "Build Phases", click on "Embed Frameworks". If I'm correct, "libssh2.framework" and "openssl.framework" are absent from the list. Add them, so it looks like this: 
![quickfix](https://user-images.githubusercontent.com/8070736/37731385-3698b2ca-2d42-11e8-8773-6a6cf203dc78.png)

Let me know if this helps.

**Comment by louisdh** [comment]
Got it working now, kinda weird that the OpenTerm project wasn't embedding libssh2 and openssl, but it's working now!

**Comment by louisdh** [comment]
Fixed in 95fc93c7beea836ea885b197c08c1490f5b7dbe0

**Comment by louisdh** [comment]
Added a unit test for this issue: c0a53c104c503d142c3dc7c6600ca78fe328a47c, so we'll know if this ever breaks again 🙂
